### PR TITLE
Streamline query/analysis of nearby sources in Gaia

### LINKED ISSUE #308: Separate some Kowalski queries from main feature generation script
For fields/ccd/quads near the galactic plane with many sources, the `drop_close_bright_stars` portion of our feature generation script (querying Gaia EDR3 for sources within 300 arcsec of each ZTF source) can take several hours. This is a significant amount of time where GPUs are not needed, but are currently being used anyway. If we ran this part of feature generation separately on CPUs and saved the results, we could reduce compute cost and dedicate more of the 48-hour maximum GPU walltime to period finding.

### PR DESCRIPTION
**Pull request by bfhealy**
This PR reduces the amount of time the feature generation script takes to query/identify nearby sources in Gaia that are too close to ZTF sources. Only one Gaia cone search is performed across the entire quadrant (as defined by the most extreme coordinates among sources on the list + a default 300 arcsec cushion).  In addition, a ~10x performance increase is realized by using `astropy.coordinates.angular_separation()` instead of `SkyCoord.separation()` to compute angular separations. This requires that latitudes/longitudes be calculated before looping over ZTF sources.

For the most crowded field among the 20 test fields (488), the entire process now runs in ~4 min, improving from a time of multiple hours. Closes #308

### CODE REVIEW

File: tools/generate_features.py [hunk 1]
```diff
@@ -88,174 +86,142 @@ def drop_close_bright_stars(
...
-                    "filter": {},
-                },
-            }
+    q = kowalski_instance.query(query)
 
-            q = kowalski_instance.query(query)
-            gaia_results = q['data'][catalog]
-            gaia_results_dct.update(gaia_results)
+    gaia_results = q.get('data')
+    gaia_results_dct.update(gaia_results['Gaia_EDR3'])
```
Comment (mcoughlin) [inline tools/generate_features.py:143]: Maybe make the chosen data release a variable?

File: tools/generate_features.py [hunk 2]
```diff
@@ -88,174 +86,142 @@ def drop_close_bright_stars(
...
-                xmatch_coord = SkyCoord(xmatch_ra, xmatch_dec, unit=['deg', 'deg'])
-            else:
-                xmatch_coord = SkyCoord(
-                    ra_geojson + 180.0, dec_geojson, unit=['deg', 'deg']
-                )
+    if len(query_result) > 0:
+        source_coords = np.array(
+            [x['radec_geojson']['coordinates'] for x in id_dct.values()]
+        )
+        source_coords[:, 0] += 180.0
```
Comment (mcoughlin) [inline tools/generate_features.py:154]: Should this be np.mod()?

File: tools/generate_features.py [hunk 3]
```diff
@@ -88,174 +86,142 @@ def drop_close_bright_stars(
...
+            single_result = query_result.copy()
+
+            Coords = np.copy(Query_Coords)
+
+            val = id_dct[id]
+            ra_geojson, dec_geojson = val['radec_geojson']['coordinates']
+
+            # Compute separations in radians, convert to arcsec
+            # ~10x faster than SkyCoord.separation() if lon/lat is calculated out of loop
+            all_separations = angular_separation(lon1[i], lat1[i], lon2, lat2) * 206265
```
Comment (mcoughlin) [inline tools/generate_features.py:178]: Mention what 206265 is.

**Review by mcoughlin — APPROVED** [review]
Small comments.

File: tools/generate_features.py (hunk 1 above)
Comment (bfhealy) [reply tools/generate_features.py]: Agreed!

File: tools/generate_features.py (hunk 3 above)
Comment (bfhealy) [reply tools/generate_features.py:178]: Revised the comment above.

File: tools/generate_features.py (hunk 2 above)
Comment (bfhealy) [reply tools/generate_features.py:154]: This line and line 162 address the [-180, 180] range of geojson RA values. I think it should stay as-is - an added constant to get to a [0, 360] range.

**Pull request merged**